microstream-rest/src/test/groovy/io/micronaut/microstream/rest/MicrostreamRestControllerSpec.groovy

```diff
@@ -0,0 +1,146 @@
+
+    @Override
+    Map<String, String> getProperties() {
+        [
+                "microstream.storage.people.root-class": People.class.name,
+                "microstream.storage.people.storage-directory": new File(tempDir, "people").absolutePath,
+                "microstream.storage.towns.root-class": Towns.class.name,
+                "microstream.storage.towns.storage-directory" : new File(tempDir, "towns").absolutePath,
+        ]
+    }
+
+    void 'file statistics exists'() {
+        when:
+        def statistics = stats()
+
+        then:
+        statistics.creationTime
+        !statistics.channelStatistics.empty
+
+        when:
+        statistics = stats('towns')
+
+        then:
+        statistics.creationTime
+        !statistics.channelStatistics.empty
+    }
+
+    void 'dictionary exists'() {
+        when:
+        def dictionary = dictionary()
+
+        then:
+        dictionary.contains('primitive 8 bit integer signed')
+    }
+
+    void 'Object traversal works as expected'() {
+        given:
+        def manager = beanContext.getBean(StorageManager, Qualifiers.byName("people"))
+        def people = manager.root().people
+        people << "Tim"
+        people << "Sergio"
+        manager.store(people)
+
+        when:
+        def root = root()
+
+        then:
+        root.name == 'ROOT'
+        root.objectId != null
+
+        when:
+        def obj = object(root.objectId)
+
+        then:
+        obj.objectId == root.objectId
+        obj.data.size() == 1
+
+        when:
+        String id = obj.data[0]
+        obj = object(id)
+
+        then:
+        obj.objectId == id
+        obj.data.size() == 1
+        // Which contains 2 items
+        obj.data[0].size() == 2
+
+        when:
+        String timId = obj.data[0][0]
+        String sergioId = obj.data[0][1]
+
+        def tim = object(timId)
+        def sergio = object(sergioId)
+
+        then:
+        tim.objectId == timId
+        tim.data.size() == 1
+        tim.data[0] == "Tim"
+
+        and:
+        sergio.objectId == sergioId
+        sergio.data.size() == 1
+        sergio.data[0] == "Sergio"
+    }
+
+    @NonNull
+    private ViewerStorageFileStatistics stats(String manager = 'people') {
+        read("/microstream/$manager/maintenance/filesStatistics", ViewerStorageFileStatistics)
+    }
+
+    @NonNull
+    private RootObject root(String manager = 'people') {
+        read("/microstream/$manager/root", RootObject)
+    }
+
+    @NonNull
+    private String dictionary(String manager = 'people') {
+        httpClient.toBlocking().retrieve("/microstream/$manager/dictionary")
+    }
+
+    @NonNull
+    private ViewerObjectDescription object(String id, String manager = 'people') {
+        return read("/microstream/$manager/object/$id", ViewerObjectDescription)
+    }
+
+    @NonNull
+    private <T> T read(String path, Class<T> clazz) {
+        objectMapper.readValue(httpClient.toBlocking().retrieve(path), clazz)
```

Comment:
You should replace this 

`objectMapper.readValue(httpClient.toBlocking().retrieve(path), clazz)`

with: 

`httpClient.toBlocking().retrieve(path, clazz)`